packages/server/src/routes/provider/resource/draft/[_id].svelte

```diff
@@ -2,56 +2,53 @@
   import { ResourceSchedule } from "@upswyng/common";
 
   export async function preload({ params, query }, { user }) {
-    if (!user || !user.isAdmin) {
-      this.error(401, "You must be an admin to access this page.");
+    let draftResource = null;
+    if (!user.isAdmin) {
```

Comment:
```suggestion
    if (!user?.isAdmin) {
```
If we supply a generic message, suggested below, whenever a draft can't be retrieved, then this could handle a user accessing a draft when they aren't logged in.

It's not a huge deal, but would avoid this weird error:
<img width="401" alt="Screen Shot 2021-01-21 at 9 52 38 PM" src="https://user-images.githubusercontent.com/6598084/105448211-1b894300-5c33-11eb-8a5f-9e187df38f5a.png">

Reply:
Ahh good catch. Is there a route through the UI I missed that allows a non logged in user to access drafts? We can do better than a 500 in any case!

If we ditch the check as I mentioned below, then I think what you recommended is a good fix (well, this specific line would be deleted, but the `user.isAdmin` below in the return).

If we keep the check, then I think
```js      
if(!user) {
  this.error(401, "You don't have permission to view this page");
} 
```
at the very top would be the correct way to handle it.

Reply:
Ya I think we should continue checking the user is accessing one of their own drafts, so your proposal of:

1. check if there is a user
2. check that the user can access the draft
3. show the things

Works for me 👍

Reply:
05a4101d9876a22cd7367313a41820fc88888a64
Alright, added that `!user` check to the top.
Also realized that I needed `return` calls below each `this.error` call... not a huge issue, but gives the correct error message that way.
I think we should be good to go now!